src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/InMemoryTests/ServersTests.InMemory.cs

```diff
@@ -0,0 +1,526 @@
+﻿//
+// Copyright (c) Microsoft.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using AnalysisServices.Tests.Helpers;
+using Microsoft.Azure;
+using Microsoft.Azure.Management.Analysis;
+using Microsoft.Azure.Management.Analysis.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit;
+
+namespace AnalysisServices.Tests.InMemoryTests
+{
+    public class InMemoryServersTests
+    {
+        [Fact]
+        public void ServerCreateAsyncValidateMessage()
+        {
+            var acceptedResponse = new HttpResponseMessage(HttpStatusCode.Created)
+            {
+                Content = new StringContent(AnalysisServicesTestUtilities.GetDefaultCreatedResponse("InProgress"))
+            };
+
+            acceptedResponse.Headers.Add("x-ms-request-id", "1");
+            acceptedResponse.Headers.Add("Location", @"http://someLocationURL");
+
+            var okResponse = new HttpResponseMessage(HttpStatusCode.OK)
```

Comment:
I am working on it and will submit a separate PR for that. SPN based authentication doesn;t work so I need to figure out how to do username based auth. I want to make progress on powershell code though. thats why I submitted this.